### DISCUSSION TITLE
PHP?

### DISCUSSION THREAD

**Discussion by zaunere** [facebookresearch/faiss#3729] [Ideas]
Hello,

Any PHP support planned, ideally taking advantage of threading/coroutines and GPU?  FFI?

Thanks!

Hans

**Comment by mdouze** [comment]
We are considering adding support for one other language (in addition to Python) but unfortunately it will most likely not be PHP because PHP is rarely used for numerical computations.

**Reply by zaunere** [reply]
Why not?  PHP is a great "glue" language - With FFI/extensions, projects like go-php and phpy, direct bindings to every libev* library and tons of other stuff, numpowr, RubixML, and stuff like amphp/swoole to make async coding normal, and much more, it's extremely utilitarian.  It's a long way from mod_php (which made Facebook :)

**Reply by alexanderguzhva** [reply]
We welcome the contributions!
